controllers/tag.js

```diff
@@ -59,3 +59,50 @@ exports.deleteBySlug = async (req, res) => {
     msg: 'ok',
   });
 };
+
+/**
+ * @route POST "/api/tag/:slug/follow"
+ */
+exports.follow = async (req, res) => {
```

Comment:
I know this code works but there is an easier way to write this, since if you come back to it in future you may not understand it. It's very confusing as to what you're doing here.

Do comment your code in blocks like

```javascript
// Checking if user already exists in followers or not
```

Also
1. Avoid using `==` in Javascript, because it is terrible. Always use `===`.
2. A better way to write if doc doesn't exist is `if (!doc) { }` (if no doc)
3. Use `res.json` everywhere
4. For checking if elem exists in array, `.includes` is easier to read than `.indexOf`
5. I think ._id gives ObjectID instance, whereas .id gives the string representation. Try it out.
6. Send relevant error status codes if action isn't performed, like 403 would work here.